cmd/utils.go

```diff
@@ -1139,3 +1139,43 @@ func newTLSConfig(getCert certs.GetCertificateFunc) *tls.Config {
 	}
 	return tlsConfig
 }
+
+// GetMinMaxTimeInfo - return time infos having min and max current time in given slice
+func GetMinMaxTimeInfo(times []madmin.TimeInfo) (min madmin.TimeInfo, max madmin.TimeInfo) {
+	min = times[0]
+	max = times[0]
+	for _, t := range times {
+		if len(t.Error) > 0 {
+			continue
+		}
+
+		if t.CurrentTime.After(max.CurrentTime) {
+			max = t
+		}
+
+		if t.CurrentTime.Before(min.CurrentTime) {
+			min = t
+		}
+	}
+	return min, max
+}
+
+// AnalyzeTimeInfo - gets time from all nodes and returns a
+// Sys Error if there is a significant deviation among them.
+func AnalyzeTimeInfo(ctx context.Context) madmin.SysErrors {
+	localTimeInfo := madmin.GetTimeInfo(ctx, globalLocalNodeName)
+	timeInfos := append(globalNotificationSys.GetTimeInfo(ctx), localTimeInfo)
+	min, max := GetMinMaxTimeInfo(timeInfos)
+	diff := max.CurrentTime.Sub(min.CurrentTime)
+	if diff > 10*time.Second {
+		return madmin.SysErrors{
+			NodeCommon: madmin.NodeCommon{Addr: max.Addr},
+			Errors: []string{
+				fmt.Sprintf("Time mismatch on servers. Max time %v on %s is away from min time %v on %s by %v",
+					max.CurrentTime, max.Addr, min.CurrentTime, min.Addr, diff),
+			},
+		}
+	}
+
+	return madmin.SysErrors{}
+}
```

Comment:
A logic of something like this might be nice

```go
			roundtrip := time.Since(sent)
			// Add 50% of the roundtrip.
			delta := time.Since(resp.Time.Add(roundtrip / 2))
			if delta < 0 {
				delta = -delta
			}
			if delta > time.Second {
				return fmt.Errorf("host %v time delta too big (%v). Roundtrip took %v. Synchronize clock on client and retry", host, delta.Round(time.Millisecond), roundtrip.Round(time.Millisecond))
			}
			return nil
```